README.md

```diff
@@ -19,14 +19,188 @@ The cleaned must-gather can then be found in the `must-gather-output-cleaned` fo
-
-In any case, if you want to obfuscate your domain names (e.g. DNS entries), then you have to adjust the list of `domainNames` to include yours.
+under [examples/openshift_default.yaml](examples/openshift_default.yaml). If you want to obfuscate your domain names (e.g. DNS entries), then you have to adjust the list of `domainNames` to include yours.
 
 In case you don't need networking or SDN information in the must-gather, you can run the configuration under [examples/openshift_omit_network.yaml](examples/openshift_omit_network.yaml). 
 This will ignore the largest files that also take a long time to obfuscate.
 
-The fully supported schema defined in [JSON schema](https://json-schema.org/)
-can be found in [schema.json](pkg/schema/schema.json) along with examples and documentation. A more browsable
+The configuration is explained along examples below, a fully supported schema defined in [JSON schema](https://json-schema.org/)
+can be found in [schema.json](pkg/schema/schema.json) along with more examples and documentation. A more browsable
 alternative can be found here
-on [json-schema.app](https://json-schema.app/view/%23?url=https%3A%2F%2Fraw.githubusercontent.com%2Fopenshift%2Fmust-gather-clean%2Fmain%2Fpkg%2Fschema%2Fschema.json).
\ No newline at end of file
+on [json-schema.app](https://json-schema.app/view/%23?url=https%3A%2F%2Fraw.githubusercontent.com%2Fopenshift%2Fmust-gather-clean%2Fmain%2Fpkg%2Fschema%2Fschema.json).
+
+## Obfuscation
+
+The configured obfuscation logic will be applied to each non-omitted file and on a line-by-line basis.
+Let's start with a very minimal obfuscation example:
+
+```
+config:
+  obfuscate:
+  - type: MAC
+```
+
+This configuration applied on a `must-gather` will detect all MAC addresses recursively in all the files. 
+Since obfuscation is about replacing the found information, the above will simply replace the found MAC address with `xx:xx:xx:xx:xx:xx`. We call this a `Static` replacement, which is the default for all types of obfuscators.
+
+There is another type of replacement called `Consistent`, that can be configured like this:
+
+```
+config:
+  obfuscate:
+  - type: MAC
+    replacementType: Consistent
+```
+
+This will detect all MAC addresses and replace them with a "consistent" identifier that looks like this `xxx-mac-000001-xxx`. 
+Let's say one of your eth interfaces has the mac address `52:54:00:5e:ee:c6` and was logged, then `must-gather-clean`  will guarantee that it will always be assigned the same obfuscated consistent identifier across all files in a must-gather. 
+That primarily helps our support and engineers to ensure we can still understand and reproduces challenges that you were facing without putting your classified information at risk.
+
+Let's look at another obfuscation type named `IP`. This type can be used to clean IP addresses, we support both IPv4 and IPv6 except for local interfaces (`127.0.0.1`, `0.0.0.0` and `::1`) that will be preserved.
```

Comment:
Same comment as above. To represent as a title instead of 'Let's say' E.g Example for obfuscation type named IP followed by the content from next line.